lib/IAMClient.js

```diff
@@ -172,10 +172,10 @@ class VaultClient {
             Version: '2010-05-08',
         };
         if (typeof options.marker !== 'undefined') {
-            const marker = parseInt(options.marker, 10);
-            assert.notStrictEqual(isNaN(marker), true,
+            const marker = options.marker;
+            assert.notStrictEqual(isNaN(marker) || marker === '', true,
```

Comment:
minor: I would rather simply `assert(!(condition), ...)` since the expression passed is a conditional already.